containers-tests/tests/IntMapValidity.hs

```diff
@@ -46,9 +48,12 @@ commonPrefix t =
   case t of
     Nil -> True
     Tip _ _ -> True
-    b@(Bin p _ l r) -> all (sharedPrefix p) (keys b) && commonPrefix l && commonPrefix r
+    b@(Bin p l r) ->
+      let px = unPrefix p
+          prefix = px .&. (px-1)
+      in all (sharedPrefix prefix) (keys b) && commonPrefix l && commonPrefix r
   where
-    sharedPrefix :: Prefix -> Int -> Bool
+    sharedPrefix :: Int -> Int -> Bool
     sharedPrefix p a = p == p .&. a
 
 -- Invariant: In Bin prefix mask left right, left consists of the elements that
```

Comment:
This comment (and any others mentioning the old structure) should be changed. I won't require perfection, but try to give it a once-over.

Reply:
These edits were actually a bit silly. It makes sense to redo this validation code, updated now.
I also can't find any more mentions like `Bin prefix mask` (except in the context of `IntSet`) but let me know if I missed any.